rafthttp/transport.go

```diff
@@ -85,17 +93,21 @@ type transport struct {
 	raft         Raft
 	serverStats  *stats.ServerStats
 	leaderStats  *stats.LeaderStats
+	v3demo       bool
 
-	mu      sync.RWMutex         // protect the term, remote and peer map
-	term    uint64               // the latest term that has been observed
-	remotes map[types.ID]*remote // remotes map that helps newly joined member to catch up
-	peers   map[types.ID]Peer    // peers map
+	mu         sync.RWMutex         // protect below fields
+	term       uint64               // the latest term that has been observed
+	remotes    map[types.ID]*remote // remotes map that helps newly joined member to catch up
+	peers      map[types.ID]Peer    // peers map
+	urlPickers map[types.ID]*urlPicker
```

Comment:
> each peer has its own picker

yes

I use urlPickers because fetching it is inconvenient:
1. it exists in peer struct in most cases, and otherwise it exists in remote struct(which happens for newly-joined member)
2. it needs to pass peer and remote map into snapshotHub, which sounds heavy

I may miss the importance of some concepts. Do you perfer to fetch url picker from peer/remote map?